tests/PS09.md

```diff
@@ -0,0 +1,595 @@
+
+The test connects to CAN server and invokes name resolve API:
+
+  * `/api/v0/name/resolve?arg=$name`, where $name is set to CAN ID of the test's identity
+
+and reads the response:
+
+  `{ "Path": "$path" }`
+
+Then the test downloads $path object from CAN server.
+
+
+###### Step 4:
+
+The test sends *CanPublishIpnsRecordRequest*:
+
+  * `Message.id := 5`
+  * `CanPublishIpnsRecordRequest.record.value := $ObjectPath1`
+  * `CanPublishIpnsRecordRequest.record.validityType := ValidityType.EOL`
+  * `CanPublishIpnsRecordRequest.record.validity` is set to a binary representation of a RFC3339 string of a future timestamp.
+  * `CanPublishIpnsRecordRequest.record.sequence := 2`
+  * `CanPublishIpnsRecordRequest.record.ttl := 6000000000`
+  * `CanPublishIpnsRecordRequest.record.signature` is set to a valid signature of this IPNS record.
+
+and reads the response.
+
+The test connects to CAN server and invokes name resolve API:
+
+  * `/api/v0/name/resolve?arg=$name`, where $name is set to CAN ID of the test's identity.
+
+and reads the response:
+
+  `{ "Path": "$path" }`
+
+
+
+###### Step 5:
+
+The test sends *CanStoreDataRequest*:
+
+  * `Message.id := 6`
+  * `CanStoreDataRequest.data.hostingServerId` is set to profile server's network ID.
+  * `CanStoreDataRequest.data.keyValueList := 
+     [
+       { key := "key1", stringValue := "value 1" },
+       { key := "key2", uint32Value := 3 },
+       { key := "key3", boolValue := false },
+       { key := "keyX", binaryValue is set to 500,000 bytes 0x40 },
+     ]`
+  * `$IdentityData2 := CanStoreDataRequest.data`
+ 
+and reads the response in form of *CanStoreDataResponse*:
+
+  * `$ObjectHash2 := CanStoreDataResponse.hash`
+
+Then it sends *CanPublishIpnsRecordRequest*:
+
+  * `Message.id := 7`
+  * `$ObjectPath2 := "/ipfs/" + Base58($ObjectHash2)`
+  * `CanPublishIpnsRecordRequest.record.value := $ObjectPath2`
+  * `CanPublishIpnsRecordRequest.record.validityType := ValidityType.EOL`
+  * `CanPublishIpnsRecordRequest.record.validity` is set to a binary representation of a RFC3339 string of a future timestamp.
+  * `CanPublishIpnsRecordRequest.record.sequence := 3`
+  * `CanPublishIpnsRecordRequest.record.ttl := 6000000000`
+  * `CanPublishIpnsRecordRequest.record.signature` is set to a valid signature of this IPNS record.
+
+and reads the response. 
+
+
+
+###### Step 6:
+
+The test waits 10 seconds to allow TTL of IPNS record to expire.
+
+The test connects to CAN server and invokes name resolve API:
+
+  * `/api/v0/name/resolve?arg=$name`, where $name is set to CAN ID of the test's identity.
+
+and reads the response:
+
+  `{ "Path": "$path" }`
+
+Then the test downloads $path object from CAN server.
+
+
+###### Step 7:
+
+The test sends *CanStoreDataRequest*:
+
+  * `Message.id := 8`
+  * `CanStoreDataRequest.data := null`
+ 
+and reads the response in form of *CanStoreDataResponse*:
+
+
+Then the test invokes CAN API pin rm:
+
+  * `api/v0/pin/rm?arg=$path`, where $path is $ObjectPath2. 

```

Comment:
This is test of profile server functionality, I want to verify that PS did its job.